File: SysVentasG2/src/main/java/pe/edu/upeu/dao/PersonaDao.java
package pe.edu.upeu.dao;

import pe.edu.upeu.crud.AppCrud;
import pe.edu.upeu.modelo.PersonaTO;
import pe.edu.upeu.util.LeerArchivo;
import pe.edu.upeu.util.LeerTeclado;
import pe.edu.upeu.util.UtilsX;

public class PersonaDao extends AppCrud{
    LeerTeclado leerT=new LeerTeclado();
    UtilsX util=new UtilsX();
    LeerArchivo leerA;
    PersonaTO pTO;
    final static String TABLA_PERSONA="Persona.txt";

    public void registrarPersona() {
        pTO=new PersonaTO();
        pTO.setDni(leerT.leer("", "Ingrese su DNI"));
        pTO.setNombre(leerT.leer("", "Ingrese nombre:"));
        pTO.setApellidos(leerT.leer("", "Ingrese apellidos"));
        pTO.setCelular(leerT.leer("","Ingrese su celular"));
        pTO.setCorreo(leerT.leer("", "Ingrese correo"));
        leerA=new LeerArchivo(TABLA_PERSONA);
        agregarContenido(leerA, pTO);
    }

    public void listarPersona() {
        leerA=new LeerArchivo(TABLA_PERSONA);
        Object[][] data=listarContenido(leerA);
        System.out.printf("%-10s %-10s %-10s %-10s %-10s%n", 
        "DNI", "Nombre","Apellidos",
         "Celular", "Correo");
        for (int i = 0; i < data.length; i++) {
            System.out.printf("%-10s %-10s %-10s %-10s %-10s%n",
            data[i][0], data[i][1],
            data[i][2], data[i][3],data[i][4]);
        }
    }

    public void editarPersona() {
        leerA=new LeerArchivo(TABLA_PERSONA);
        pTO=new PersonaTO();
        String dnix=leerT.leer("", "Busque con su DNI");
        pTO.setCelular(leerT.leer("","Ingrese nuevo celular"));
        pTO.setCorreo(leerT.leer("", "Ingrese su nuevo correo"));
        editarRegistro(leerA, 0, dnix, pTO);
    }

    public void eliminarPersona() {
        leerA=new LeerArchivo(TABLA_PERSONA);
        String dnix=leerT.leer("", "Busque con su DNI");
        eliminarRegistros(leerA, 0, dnix);
    }

}

File: ProyectoIntegrador/pe/edu/upeu/dao/PersonaDao.py

from pe.edu.upeu.modelo.PersonaTO import PersonaTO
from pe.edu.upeu.util.LeerArchivo import LeerArchivo


class PersonaDao:
    global nombrea
    nombrea="Persona.txt"

    def crearPersona(self):
        pTo=PersonaTO("","","","","")
        pTo.dni=str(input("Ingrese su DNI:"))
        pTo.nombre=input("Ingrese su nombre:")
        pTo.apellidos = input("Ingrese su apellido:")
        pTo.celular = input("Ingrese su Celular:")
        pTo.correo = input("Ingrese su correo:")

        aq=LeerArchivo(nombrea)
        contenido=""
        for campo, valor in pTo.__dict__.items():
            if campo=="correo":
                contenido = "".join((contenido, valor))
            else:
                contenido = "".join((contenido, valor + "\t"))        
        aq.agregarContenidoArchivo(contenido)

    def listarPersona(self):
        aq = LeerArchivo(nombrea)
        data=aq.leerContenidoArchivo()
        for d in data:
            print(d)
            
    def editarPersona(self):
        aq = LeerArchivo(nombrea)
        modeloP=PersonaTO("","","")
        dnix=str(input("Ingrese DNI:"))
        modeloP.apellidos=input("Ingrese nuevo apellido:")
        data = aq.leerContenidoArchivo()
        matriz = []
        for d in data:
            separado = d.split("\t")
            matriz.append(separado)
        print(matriz)
        for f in range(len(matriz)):
            ix=0
            if(matriz[f][2]==dnix):
                print("entro aqui")
                verx=modeloP.__dict__.items()
                print(verx)
                for campo, valor in modeloP.__dict__.items():
                    print(f"i:{ix} {valor}")
                    if valor!="":
                        matriz[f][ix] = valor
                    ix+=1

        for x in range(len(matriz)):
            contenido = ""
            for y in range(len(matriz[0])):
                if y == len(matriz[0])-1:
                    contenido = "".join((contenido, str(matriz[x][y])+"\n"))
                else:
                    contenido = "".join((contenido, str(matriz[x][y]) + "\t"))
            if x==0:
                aq = LeerArchivo(nombrea)
                aq.escribirArchivo(contenido)
            else:
                aq = LeerArchivo(nombrea)
                aq.agregarContenidoArchivo(contenido)



    def eliminarPersona(self):
        aq = LeerArchivo(nombrea)

        pass
